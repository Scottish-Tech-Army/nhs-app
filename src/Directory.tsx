import React from "react";
import "./App.css";
import { BoxTemplate } from "./model/StorageTypes";
import { TRAUMA_TOWER_TEMPLATE } from "./model/TraumaTower";
import { useNavigate } from "react-router-dom";
import { ReactComponent as EditIcon } from "./icons/edit.svg";
import Navbar from "./Navbar";

function Directory() {
  let navigate = useNavigate();

  const getBoxes = (boxTemplate: BoxTemplate) => {
    const result = [];
    for (let boxIndex = 1; boxIndex <= boxTemplate.count; boxIndex++) {
      result.push(
        <div
          className="box"
          key={`${boxTemplate.boxTemplateId}/${boxIndex}`}
          onClick={() =>
            navigate(`/box/${boxTemplate.boxTemplateId}/${boxIndex}`)
          }
        >
          <EditIcon />
          <div className="display-name">{`${boxTemplate.name} - Box ${boxIndex}`}</div>
        </div>
      );
    }
    return result;
  };

  return (
    <div className="directory">
      <header>
        <h1>{TRAUMA_TOWER_TEMPLATE.name}</h1>
      </header>
      <main>
        <div className="scroll">
          {TRAUMA_TOWER_TEMPLATE.boxes.map((boxTemplate) =>
            getBoxes(boxTemplate)
          )}
        </div>
      </main>
      <footer>
        <Navbar />
      </footer>
    </div>
  );
}

export default Directory;
