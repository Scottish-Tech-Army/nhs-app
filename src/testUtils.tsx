import React from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import type { AppStore, RootState } from "./data/store";
import boxContentsReducer, { initialState } from "./data/BoxContentsSlice";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProvider(
  component: React.ReactElement,
  {
    preloadedState = { boxContents: initialState },
    store = configureStore({
      reducer: { boxContents: boxContentsReducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {

  return {
    store,
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
}
