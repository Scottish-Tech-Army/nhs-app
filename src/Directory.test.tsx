/* eslint-disable testing-library/no-node-access */
import { screen } from "@testing-library/react";
import Directory from "./Directory";
import "@testing-library/jest-dom";
import React from "react";
import { renderWithProvider } from "./testUtils";

describe("Directory", () => {
  it("renders a home page", async () => {
    renderWithProvider(<Directory />);

    expect(screen.getByText("Directory")).toBeDefined();

    [
      "Trauma Tower 1",
      "Trauma Tower 2",
      "Storage area",
    ].forEach((title) => {
      expect(screen.getByText(title)).toBeDefined();
    });

    const summaryLink = await screen.findByRole("link", { name: "summary" });
    expect(summaryLink).toHaveAttribute("href", "/summary");
  });

  it("renders correctly", () => {
    const { container } = renderWithProvider(<Directory />);

    expect(container).toMatchSnapshot();
  });

  it("navigates to storage area page", async () => {
    const { user, history } = renderWithProvider(<Directory />);

    expect(screen.getByText("Directory")).toBeDefined();
    expect(history.location.pathname).toEqual("/");

    await user.click(await screen.findByText("Trauma Tower 2"));

    expect(history.location.pathname).toEqual("/area/1");
  });
});
