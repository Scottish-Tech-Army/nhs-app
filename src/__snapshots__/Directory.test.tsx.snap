// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory renders correctly 1`] = `
<div>
  <div
    class="directory"
  >
    <header>
      <h1>
        Directory
      </h1>
    </header>
    <main>
      <div
        class="scroll"
      >
        <div
          class="single-storage-area"
        >
          <svg>
            edit.svg
          </svg>
          <div
            class="display-name"
          >
            Trauma Tower 1
          </div>
        </div>
        <div
          class="single-storage-area"
        >
          <svg>
            edit.svg
          </svg>
          <div
            class="display-name"
          >
            Trauma Tower 2
          </div>
        </div>
        <div
          class="single-storage-area"
        >
          <svg>
            edit.svg
          </svg>
          <div
            class="display-name"
          >
            Backup Area
          </div>
        </div>
      </div>
    </main>
    <footer>
      <nav>
        <a
          aria-current="page"
          aria-label="directory"
          class="navicon active"
          href="/"
        >
          <svg>
            home.svg
          </svg>
          <div>
            Directory
          </div>
        </a>
        <a
          aria-label="summary"
          class="navicon"
          href="/summary"
        >
          <svg>
            checklist.svg
          </svg>
          <div>
            Summary
          </div>
        </a>
        <div
          aria-label="log out"
          class="navicon"
          id="auth-signout"
        >
          <svg>
            logout.svg
          </svg>
          <div>
            Log Out
          </div>
        </div>
      </nav>
    </footer>
  </div>
</div>
`;
